src/osx/core/colour.cpp

```diff
@@ -13,127 +13,244 @@
 #include "wx/colour.h"
 
 #ifndef WX_PRECOMP
-    #include "wx/gdicmn.h"
+#include "wx/gdicmn.h"
 #endif
 
 #include "wx/osx/private.h"
 
+CGColorSpaceRef wxMacGetGenericRGBColorSpace();
+
+wxColorRefData::~wxColorRefData()
+{
+}
+
+class wxCGColorRefData : public wxColorRefData
+{
+public:
+    wxCGColorRefData(CGFloat r, CGFloat g, CGFloat b, CGFloat a = 1.0);
+
+    wxCGColorRefData(CGFloat components[4]);
+
+    wxCGColorRefData(CGColorRef);
+
+    wxCGColorRefData(const wxCGColorRefData& other);
+
+    virtual ~wxCGColorRefData()
+    {
+    }
+
+    virtual bool IsOk() const  wxOVERRIDE{ return m_cgColour != NULL; }
+
+    virtual CGFloat Red() const wxOVERRIDE { return m_red; }
+    virtual CGFloat Green() const wxOVERRIDE { return m_green; }
+    virtual CGFloat Blue() const wxOVERRIDE { return m_blue; }
+    virtual CGFloat Alpha() const wxOVERRIDE { return m_alpha; }
+
+    CGColorRef GetCGColor() const wxOVERRIDE { return m_cgColour; }
+
+    virtual wxColorRefData* Clone() const wxOVERRIDE { return new wxCGColorRefData(*this); }
+
+private:
+    void Init(CGFloat components[4]);
+
+    wxCFRef<CGColorRef> m_cgColour;
+
+    CGFloat m_red;
+    CGFloat m_blue;
+    CGFloat m_green;
+    CGFloat m_alpha;
+
+    wxDECLARE_NO_ASSIGN_CLASS(wxCGColorRefData);
+};
+
+wxCGColorRefData::wxCGColorRefData(CGFloat r, CGFloat g, CGFloat b, CGFloat a)
+{
+    CGFloat components[4];
+    components[0] = r;
+    components[1] = b;
+    components[2] = g;
+    components[3] = a;
+
+    Init(components);
+}
+
+wxCGColorRefData::wxCGColorRefData(CGFloat components[4])
+{
+    Init(components);
+}
+
+wxCGColorRefData::wxCGColorRefData(const wxCGColorRefData& other)
+{
+    m_red = other.m_red;
+    m_blue = other.m_blue;
+    m_green = other.m_green;
+    m_alpha = other.m_alpha;
+
+    m_cgColour = other.m_cgColour;
+}
+
+void wxCGColorRefData::Init(CGFloat components[4])
+{
+    m_red = components[0];
+    m_blue = components[1];
+    m_green = components[2];
+    m_alpha = components[3];
+
+    CGColorRef cfcol;
```

Comment:
It's better to initialize when declaring instead of using a separate assignment. It could also be made `const` then.